packages/content/src/types.ts

```diff
@@ -0,0 +1,137 @@
+import { IItem, IGroup } from "@esri/arcgis-rest-types";
+
+export type HubType = 'document' | 'dataset' | 'map' | 'app' | 'site' | 'initiative' | 'event' | 'template';
+
+// Based on the [Hub Content UI Inventory](https://app.lucidchart.com/invitations/accept/dab82a25-2ed9-4f63-8075-3db8873087e3)
+// Attempting a simple _Hungarian Notation_ for attributes suffix: `*Date`, `*Id`, `*Number`. Plural (e.g. `tags`) is an array.
+
+export interface IContent extends IItem {
+  
+  // Inherited metadata (noting relevant subset of all item attributes)
+  // id: string; // item.id
+  // title: string; // item.title
+  // snippet: string; // item.snippet
+  // description: string; // item.description
+  // categories?: Array<string>; // item.categories
+  // tags: Array<string>; // item.tags  
+  // culture: string; // written locale e.g. en, es, cn
+  // url?: string;
+  // size: number;
+
+  // Derived metadata
+  hubType: HubType;
+  maintainer: IHubMember; // item.owner with more user metadata
+  access: { // overrides item.access with more attributes. could flatten.
+    visibility: 'private' | 'org' | 'public'; // item.access
+    permission?: 'view' | 'edit' | 'admin'; // item.itemControl
+    groups?: Array<IGroup>; // item.sharing.groups via content/users/:username/items/:id
+  };
+
+  // Explicit data information since this is a common confusion + bug report
+  createdDate: Date; // formal metadata || new Date(item.created)
+  createdDateSource: string; // description of what was used for this attribute
+  publishedDate: Date; // formal metadata || new Date(item.created)
+  publishedDateSource: string; // description of what was used for this attribute
+  updatedDate: Date; // formal metadata || new Date(item.modified)
+  updatedDateSource: string; // description of what was used for this attribute
+  
+  contentUrl: string; // Link to the raw content. item.url in most (but not all) item types
+  thumbnailUrl: string; // Full URL. item.thumbnail with host + path 
```

Comment:
`thumbnailUrl` is a good example of a derived, read-only property. There is no way to update this property and change the thumbnail. Instead you have to POST w/ a Blob payload in the `thumbnail` property, and then, the backend api will return a new name in the thumbnail property, and we can re-compute the `thumbnailUrl` property.

Reply:
again, I don't think we use the same keys as the item, in this case I propose something like `thumbnailFullUrl`

Reply:
@dbouwman what if the implementation for WRITE of `thumbnailUrl` fetched the image and pushed it via the API? Alternatively, the update would return back either quietly that it didn't use the value or an explicit "worked but..."

This is the kind of implementation detail we want to isolate from end developers. If we think that someone should be able to set the thumbnail by URL then we can add that within the update method.

@tomwayson the original attribute is `item.thumbnail`. I explicitly did not use the same key by appending the "type" via `content.thumbnailUrl`. Since we extend `IItem` there still would be `content.thumbnail`.